DevCycle/Networking/DevCycleService.swift

```diff
@@ -147,6 +151,40 @@ class DevCycleService: DevCycleServiceProtocol {
         }
     }
     
+    func saveEntity(user: DVCUser, completion: @escaping SaveEntityCompletionHandler) {
+        var saveEntityRequest = createSaveEntityRequest()
+        
+        guard let userIsAnonymous = user.isAnonymous, !userIsAnonymous else {
+            Log.error("Cannot save user data for an anonymous user!")
+            return
+        }
+        
+        let userEncoder = JSONEncoder()
+        userEncoder.dateEncodingStrategy = .iso8601
+        
+        guard let userData = try? userEncoder.encode(user) else {
+            return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
+        }
+        
+        guard let userBody = try? JSONSerialization.jsonObject(with: userData, options: .fragmentsAllowed) else {
+            return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
+        }
+        
+        let jsonBody = try? JSONSerialization.data(withJSONObject: userBody, options: .prettyPrinted)
+        
+        saveEntityRequest.httpMethod = "PATCH"
+        saveEntityRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        saveEntityRequest.addValue("application/json", forHTTPHeaderField: "Accept")
+        saveEntityRequest.addValue(config.environmentKey, forHTTPHeaderField: "Authorization")
+        Log.info("Save entity payload: \(String(data: jsonBody!, encoding: .utf8) ?? "")")
```

Comment:
this will cause an error if you try to unwrap a nil, the best practice on unwrapping values in Swift is unwrapping safely: 

```
if let body = jsonBody {
   // build the save entity request with this data object
} else {
   // something went wrong with the user data
}
```